hazelcast-sql/src/test/java/com/hazelcast/jet/sql/StaleIndexTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql;
+
+import com.hazelcast.config.IndexConfig;
+import com.hazelcast.config.IndexType;
+import com.hazelcast.map.IMap;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.Serializable;
+
+import static java.util.Collections.singletonList;
+
+public class StaleIndexTest extends SqlTestSupport {
```

Comment:
I don't know if that test tests what you really wanted, my console output shows just one plan:
```
17:06:52,030 DEBUG |test| - [CalciteSqlOptimizer] test - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Before logical opt:
LogicalCalc(expr#0..3=[{inputs}], expr#4=[_UTF-16LE'v0':VARCHAR CHARACTER SET "UTF-16LE"], expr#5=[=($t1, $t4)], __key=[$t0], $condition=[$t5])
  LogicalTableScan(table=[[hazelcast, public, m[projects=[$0, $1, $2, $3]]]])

17:06:52,190 DEBUG |test| - [CalciteSqlOptimizer] test - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] After logical opt:
FullScanLogicalRel(table=[[hazelcast, public, m[projects=[$0], filter==($1, _UTF-16LE'v0')]]])

17:06:52,248 DEBUG |test| - [CalciteSqlOptimizer] test - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] After physical opt:
FullScanPhysicalRel(table=[[hazelcast, public, m[projects=[$0], filter==($1, _UTF-16LE'v0')]]], discriminator=[0])

17:06:52,312 DEBUG |test| - [LightMasterContext] test - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Start executing light job 08f4-8b38-a440-0001, execution graph in DOT format:
digraph DAG {
	"IMap[public.m]" [localParallelism=1];
	"Project(IMap[public.m])" [localParallelism=2];
	"ClientSink" [localParallelism=1];
	"IMap[public.m]" -> "Project(IMap[public.m])" [label="isolated", queueSize=1024];
	"Project(IMap[public.m])" -> "ClientSink" [label="distributed to [127.0.0.1]:5701-partitioned", queueSize=1024];
}
````

It is ```FullScanPhysicalRel```, not index one. And after destroying a map there is no second log with calcite optimizations, so maybe the cache is not invalidated?